scripts/index-demo-page.js

```diff
@@ -61,17 +65,49 @@
         dashOptGroup.appendChild(option);
       }
     });
+
   });
   xhr.open('GET', './scripts/sources.json');
   xhr.send();
 
+  var hlsManifestXhr = new window.XMLHttpRequest();
+
+  hlsManifestXhr.addEventListener('load', function() {
+    var hlsManifest = hlsManifestXhr.responseText;
+    var option = document.createElement('option');
+
+    option.innerText = 'HLS Manifest Object Test, does not survive page reload';
+    option.value = `data:application/vnd.videojs.vhs+json,${hlsManifest}`;
+
+    manifestOptGroup.appendChild(option);
+  });
+  hlsManifestXhr.open('GET', './scripts/hls-manifest-object.json');
+  hlsManifestXhr.send();
+
+  var dashManifestXhr = new window.XMLHttpRequest();
+
+  dashManifestXhr.addEventListener('load', function() {
+    var dashManifest = dashManifestXhr.responseText;
+    var option = document.createElement('option');
+
+    option.innerText = 'Dash Manifest Object Test, does not survive page reload';
+    option.value = `data:application/vnd.videojs.vhs+json,${dashManifest}`;
+
+    manifestOptGroup.appendChild(option);
+  });
+  dashManifestXhr.open('GET', './scripts/dash-manifest-object.json');
+  dashManifestXhr.send();
+
   // all relevant elements
   var urlButton = document.getElementById('load-url');
   var sources = document.getElementById('load-source');
   var stateEls = {};
 
   var getInputValue = function(el) {
     if (el.type === 'url' || el.type === 'text' || el.nodeName.toLowerCase() === 'textarea') {
+      if (isManifestObjectType(el.value)) {
```

Comment:
These object urls are too big to save as a query param so we have to ignore them.